src/app/zone-egresses/data/index.ts

```diff
@@ -0,0 +1,93 @@
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type {
+  ZoneEgressOverview as PartialZoneEgressOverview,
+  ZoneEgress as PartialZoneEgress,
+  KDSSubscription,
+} from '@/types/index.d'
+
+type PartialZoneEgressInsight = any
+type PartialInternalZoneEgress = PartialZoneEgressOverview['zoneEgress']
+
+// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
+type InternalZoneEgress = {
+  socketAddress: string
+} & PartialInternalZoneEgress
+
+export type ZoneEgress = {
+  config: PartialZoneEgress
+  socketAddress: string
+} & PartialZoneEgress
+// end TODO
+
+// TODO(jc) currently in our manually written types ZoneEgressInsight is any therefore
+// we have no Partial*
+export type ZoneEgressInsight = {
+  connectedSubscription?: KDSSubscription
+  subscriptions: KDSSubscription[]
+}
+export type ZoneEgressOverview = PartialZoneEgressOverview & {
+  zoneEgress: InternalZoneEgress
+  zoneEgressInsight?: ZoneEgressInsight
+  state: 'online' | 'offline'
+}
+// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
+const InternalZoneEgress = {
+  fromObject: (item: PartialInternalZoneEgress): InternalZoneEgress => {
+    return {
+      ...item,
+      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+    }
+  },
+}
+
+export const ZoneEgress = {
+  fromObject: (item: PartialZoneEgress): ZoneEgress => {
+    return {
+      ...item,
+      config: item,
+      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+    }
+  },
+}
+// end TODO
+export const ZoneEgressInsight = {
+  fromObject: (item?: PartialZoneEgressInsight): ZoneEgressInsight | undefined => {
+    // if item isn't set don't even try augmenting things
+    return isSet<PartialZoneEgressInsight>(item)
+      ? ((item) => {
+        const subscriptions: KDSSubscription[] = Array.isArray(item.subscriptions) ? item.subscriptions : []
+        // figure out the connectedSubscription by looking at the connectTime
+        // and disconnectTime of the last subscription
+        const connectedSubscription = subscriptions.slice(-1).find((item) => item.connectTime?.length && !item.disconnectTime)
```

Comment:
```suggestion
        const connectedSubscription = subscriptions.first((item) => item.connectTime?.length && !item.disconnectTime)
```

Be permissive maybe

Reply:
Following our sync on this we are defo going to do this. We want to do it holistically over the entire project though so we don't do it in one place and not another (I made an issue so we also don't forget to do it https://github.com/kumahq/kuma-gui/issues/1890)

Handily some other things have also come up related to this and how we figure out `lastUpdatedTime` and some other things to do with subscriptions so that will all be thought about addresses at once everywhere.

Resolving this here for now, but there will be a quick follow up pretty soon.